dashboard/app/views/devise/registrations/_finish_sign_up.html.haml

```diff
@@ -11,9 +11,12 @@
 
   = form_for(@user, url: registration_path(@user), html: {class: "finish-signup"}) do |f|
     %p
-      = t('activerecord.attributes.user.finish_sign_up_subheader')
-      - if @user.provider.present? && @user.email.present?
-        %span= t('activerecord.attributes.user.finish_sign_up_subheader_provider', provider: t("auth.#{@user.provider}"), email: @user.email).html_safe
+      - if @user.provider.present?
+        - provider = t("auth.#{@user.provider}")
+        = t('activerecord.attributes.user.finish_sign_up_subheader_provider', provider: provider, email: @user.email).html_safe
+      - else
+        = t('activerecord.attributes.user.finish_sign_up_subheader', email: @user.email).html_safe
```

Comment:
I feel okay about relying on having `@user.email` set at this point (the user shouldn't be able to get to this point if we don't have an email for their account), but would like to hear others' feedback on this.